### PR TITLE
Uploading updated version of Vehicle Additions Pack

### PR DESCRIPTION
**Pull request by Blaze-whatdoyoumeanitsalreadytaken**
Am I doing this right?

"Able to merge" is a good thing right?

### CODE REVIEW

File: data/mods/blazemod/blaze_ammo.json [hunk 1]
```diff
@@ -12,8 +12,11 @@
         "volume" : 1,
         "weight" : 3816,
         "bashing" : 8,
+        "cutting" : 0,
```
Comment (Coolthulhu) [inline data/mods/blazemod/blaze_ammo.json]: All entries of type:
* "cutting" : 0
* "to_hit" : 0
* "pierce" : 0
* "range" : 0

Are redundant and should be removed.

File: data/mods/blazemod/blaze_ammo_recipes.json [hunk 2]
```diff
@@ -97,8 +97,8 @@
   ],
   "components": [
     [
-      [ "can_food_unsealed", 1 ],
-      [ "can_drink_unsealed", 1 ],
+      [ "can_food", 1 ],
+      [ "can_drink", 1 ],
```
Comment (Coolthulhu) [inline data/mods/blazemod/blaze_ammo_recipes.json]: This should be "can_food_unsealed" and same for "can_drink_unsealed". Otherwise it won't ever be craftable, since you'd need an empty but sealed can.

File: data/mods/blazemod/blaze_ammo_recipes.json [hunk 3]
```diff
@@ -517,57 +518,55 @@
...
-  "time": 30000,
+  "time": 20000,
   "reversible": false,
   "autolearn": true,
   "qualities":[
-    {"id":"HAMMER","level":2,"amount":1},
-    {"id":"SAW_M","level":1,"amount":1},
-    { "id" : "GLARE", "level" : 2 }
+    {"id":"HAMMER","level":1,"amount":1},
+    {"id":"CUT","level":1,"amount":1}
```
Comment (Coolthulhu) [inline data/mods/blazemod/blaze_ammo_recipes.json]: "amount":1 is redundant. Amount should only be specified if it is `>1`.

File: data/mods/blazemod/blaze_blob.json [hunk 4]
```diff
@@ -1,1238 +1,1660 @@
...
         "type": "delayed_transform",
         "transform_age": 2400,
-        "not_ready_msg": "The blob is still growing; at a phenomenal rate you might add.",
-        "msg": "The blob emits an odd gurgle; it sounds disturbingly like \"Mama!\".",
+        "not_ready_msg": "Whatever it's doing, it's not done yet.",
+        "msg": "The blob balloons to full size.",
         "moves": 0,
-        "target": "shooter"
+        "target": "gloople",
+        "target_charges": 0
```
Comment (Coolthulhu) [inline data/mods/blazemod/blaze_blob.json]: "target_charges": 0 is redundant

File: data/mods/blazemod/blaze_blob.json [hunk 5]
```diff
@@ -1,1238 +1,1660 @@
...
+    "weight": 3562,
+    "volume": 8,
     "bashing": 0,
+    "cutting": 0,
     "to_hit": -2,
+    "max_charges": 0,
+    "initial_charges": 0,
+    "charges_per_use": 0,
+    "turns_per_charge": 0,
+    "revert_to": "null",
```
Comment (Coolthulhu) [inline data/mods/blazemod/blaze_blob.json]: All of those are redundant:
* "max_charges": 0,
* "initial_charges": 0,
* "charges_per_use": 0,
* "turns_per_charge": 0,
* "revert_to": "null",

**Comment by mugling** [comment]
@kevingranade any objection to reworking this en-masse via `jq` and some `perl`?

**Comment by Blaze-whatdoyoumeanitsalreadytaken** [comment]
I'll get to work on removing the redundant entries then!

**Comment by Rivet-the-Zombie** [comment]
>"Able to merge" is a good thing right?

It is. It means your files have no conflicts with the main branch which would prevent them from being automerged.

**Comment by mugling** [comment]
This will no doubt take some time but welcome @Blaze-whatdoyoumeanitsalreadytaken

File: data/mods/blazemod/blaze_ammo_recipes.json [hunk 6]
```diff
@@ -135,7 +135,7 @@
   "reversible": false,
   "autolearn": true,
   "qualities" : [
-    {"id":"CUT","level":1,"amount":1}
+    {"id":"CUT","level":1}
```
Comment (mugling) [inline data/mods/blazemod/blaze_ammo_recipes.json]: JSON style would be all on one line where there is just one quality (like the entry above this). Correct indentation and formatting greatly helps readability

File: data/mods/blazemod/blaze_blob.json [hunk 7]
```diff
@@ -1,1238 +1,1339 @@
...
+    "name": "growing hairy mass",
+    "name_plural" : "growing hairy mass",
+    "description": "Not quite fully grown, this blob requires nourishment to fully develop.",
+    "price": 100,
+    "material": [ "flesh" ],
+    "weight": 18140,
+    "volume": 20,
+    "to_hit": -2,
+    "category" : "tools",
+    "use_ac
```
Comment (mugling) [inline data/mods/blazemod/blaze_blob.json]: Why are these entries included?

**Comment by Blaze-whatdoyoumeanitsalreadytaken** [comment]
You mean "Growing" blob entries? If they get destroyed they can regenerate after some time passes.

File: data/mods/blazemod/blaze_blob.json (hunk 7 above)
Comment (mugling) [inline data/mods/blazemod/blaze_blob.json]: Why is this included?

Comment (mugling) [inline data/mods/blazemod/blaze_blob.json]: No the web UI can't place the comment in the correct place, possibly due to the size of this PR?

I mean why are you redefining `id": "flamethrower"` and such?

Comment (Blaze-whatdoyoumeanitsalreadytaken) [inline data/mods/blazemod/blaze_blob.json]: Oh, the gas puffer (inkie) uses a custom explosion for gasoline, but needed to remove the "stream" flag. So I removed it from gasoline and added it to the individual flamethrowers.

**Comment by mugling** [comment]
You can do:
 
```
"copy-from": "whatever",
"delete": { "flags": [ "FOO" ] }
```

**Comment by Blaze-whatdoyoumeanitsalreadytaken** [comment]
Can you use "copy-from" using the same ID as the item you're replacing?

**Comment by mugling** [comment]
> Can you use "copy-from" using the same ID as the item you're replacing?

Yes, you need to also specify (the same) `id` and `type`

**Comment by Blaze-whatdoyoumeanitsalreadytaken** [comment]
Done.

File: data/mods/blazemod/blaze_blob.json [hunk 8]
```diff
@@ -1,1238 +1,1287 @@
...
-    "description": "This appears to be some form of structure constructed from bone, held together by some unknown slimy substance.  A jury-rigged slow-drip dispenser trickles some fluid you'd prefer not to think too hard about throughout the structure.",
+    "color": "light_green",
+    "name": "multiplying gelatinous mass",
+    "name_plural" : "multiplying gelatinous mass",
+    "description": "Having been fed, this blob is now rapidly multiplying into other copies of itself; extremely noisy copies!  And even worse, it's stuck to your hands until whatever it's doing is done!  Catch those blobs before they bring in every zombie for miles!",
     "price": 100,
-    "material": ["bone", "flesh"],
-    "weight": 15662,
+    "material": ["flesh"],
+    "flags": ["NO_UNWIELD", "UNBREAKABLE_MELEE", "NO_PICKUP"],
```
Comment (Coolthulhu) [inline data/mods/blazemod/blaze_blob.json]: "NO_PICKUP" flag does not exist.

File: data/mods/blazemod/blaze_blob.json [hunk 9]
```diff
@@ -1,1238 +1,1287 @@
...
+  },
+    {
+    "id": "clutter",
+    "type": "GUN",
+    "symbol": "&",
+    "color": "purple",
+    "name": "gel shooter",
+    "description": "A living blob turned into an autonomous weapon.  It trawls the ground for material, which it then strips of nutrition.  The remainder is then expelled out at significant velocity towards any nearby threats.  The amorphous mass can be shaped and attached at your touch, but the weapon itself is inert without something to control it.  It seems pliable enough to pull apart...",
+    "material": ["flesh", "bone"],
+    "flags": ["NEVER_JAMS", "MOUNTED_GUN", "NO_AMMO"],
```
Comment (Coolthulhu) [inline data/mods/blazemod/blaze_blob.json]: "NO_AMMO" flag is obsolete.

File: data/mods/blazemod/blaze_blob.json [hunk 10]
```diff
@@ -1,1238 +1,1287 @@
...
+    "weight": 11331,
+    "volume": 18,
+    "bashing": 10,
+    "to_hit": -5,
+    "ranged_damage": 5,
+    "range": 12,
+    "dispersion": 900,
+    "sight_dispersion": 150,
+    "durability": 10,
+    "burst": 1,
```
Comment (Coolthulhu) [inline data/mods/blazemod/blaze_blob.json]: Burst 1 is redundant

File: data/mods/blazemod/blaze_blob.json [hunk 11]
```diff
@@ -1,1238 +1,1287 @@
...
+    "price": 100,
+    "material": [ "flesh" ],
+    "weight": 18140,
+    "volume": 20,
+    "to_hit": -2,
+    "category" : "tools",
+    "use_action": {
+        "type": "delayed_transform",
+        "transform_age": 4800,
+        "not_ready_msg": "Whatever it's do
```
Comment (Coolthulhu) [inline data/mods/blazemod/blaze_blob.json]: Clip size smaller than burst will most likely not work.

Comment (Coolthulhu) [inline data/mods/blazemod/blaze_blob.json]: "MODE_BURST" will most likely not work

File: data/mods/blazemod/blaze_blob_recipes.json [hunk 12]
```diff
@@ -0,0 +1,923 @@
...
+  "result": "bfeed",
+  "id_suffix": "blob assisted",
+  "category": "CC_OTHER",
+  "subcategory": "CSC_OTHER_MATERIALS",
+  "skill_used": "cooking",
+  "difficulty": 6,
+  "time": 1000,
+  "reversible": false,
+  "autolearn": true,
+  "qualities" : [],
```
Comment (Coolthulhu) [inline data/mods/blazemod/blaze_blob_recipes.json]: Empty qualities field is redundant.

`"reversible": false` is redundant

File: data/mods/blazemod/blaze_blob_recipes.json [hunk 13]
```diff
@@ -0,0 +1,923 @@
...
+},{
+    "type" : "recipe",
+    "result": "gloople",
+    "id_suffix": "uncraft",
+    "category": "CC_NONCRAFT",
+    "subcategory": "CSC_NONCRAFT",
+    "difficulty": 0,
+    "time": 5000,
+    "reversible": true,
+    "autolearn": false,
```
Comment (Coolthulhu) [inline data/mods/blazemod/blaze_blob_recipes.json]: `"autolearn": false` is redundant

File: data/mods/blazemod/blaze_blob.json (hunk 11 above)
Comment (mugling) [inline data/mods/blazemod/blaze_blob.json]: Yes, entirely redundant

Comment (mugling) [inline data/mods/blazemod/blaze_blob.json]: Will work but will be capped to `clip_size`. Doesn't make any sense however as it is.

**Comment by kevingranade** [comment]
> http://smf.cataclysmdda.com/index.php?topic=11926.msg264313#msg264313

Not for new content, that's the ideal time to do large-scale adjustments like that.

**Comment by Coolthulhu** [comment]
I didn't find any gameplay problems.
There are tons of style problems, but those will be fixed with automatic tools in #17417. It will be much faster and much less work than by hand.

File: data/mods/blazemod/blaze_override.json [hunk 14]
```diff
@@ -1,139 +1,348 @@
...
+                [ "foldframe", 1 ],
+                [ "gloople", 1],
+                [ "oozle", 1],
+                [ "gray", 1],
+                [ "diamondframe", 1]
+            ]
+        ],
+        "pre_special" : "check_empty",
+        "post_special" : "done_vehicle"
+    },{
```
Comment (darwithe) [inline data/mods/blazemod/blaze_override.json]: Could someone vastly more knowledgeable than me check if this override to "Start Vehicle Construction" still causes player constructed initial parts to instant delete?
Also I seem to remember a recent commit that rendered this change obsolete with the base version of "Start Vehicle Construction" now using anything with the initial_part flag.

**Comment by Blaze-whatdoyoumeanitsalreadytaken** [comment]
Is there anything I can do to move this along? It's been two weeks since I added this, though I haven't been around long enough to know the specifics of how things are done here.

**Comment by mugling** [comment]
> Is there anything I can do to move this along?

No, the reformatted PR (#17456) is pending review. A PR of that size is exceptional.

**Comment by mugling** [comment]
Continues in #17456
